tasks/collect-atlas-params/collect-atlas-params.yaml

```diff
@@ -0,0 +1,78 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: collect-atlas-params
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that collects Atlas API configuration options from the data file.
+  params:
+    - name: dataPath
+      type: string
+      description: |
+        Path to the JSON string of the merged data containing the Atlas config.
+  workspaces:
+    - name: data
+  results:
+    - name: bombasticApiUrl
+      type: string
+      description: |
+        URL of the bombastic API.
+    - name: ssoTokenUrl
+      type: string
+      description: |
+        URL of the SSO token issuer.
+    - name: secretName
+      type: string
+      description: |
+        The kubernetes secret to use to authenticate to bombastic.
+    - name: updateComponentSBOM
+      type: string
+      description: |
+        Flag specifying whether to update the component-level SBOMs and push
+        them to Atlas
+  steps:
+    - name: collect-atlas-params
+      image:
+        quay.io/konflux-ci/release-service-utils:d320c36f3d707cd5bfe55fe783f70236c06cc2e5
+      script: |
+        #!/usr/bin/env bash
+        set -x
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "ERROR: No valid data file was provided."
+            exit 1
+        fi
+
+        atlasServer=$(jq -r '.atlas.server' "$DATA_FILE")
+        if [ "$atlasServer" = "null" ]; then
+            echo "ERROR: The JSON file does not contain the 'atlasServer' field."
```

Comment:
Would it make sense to have a default value for this? Or maybe have this whole thing optional? I'm just worried about the transition to enable this. If it's really gonna be mandatory, is everything ready now? Do all the RPAs have this defined now? Are the secrets ready on all clusters? Because some users may be using the development branch.

Reply:
This was my worry as well, but I was guided by [this discussion](https://redhat-internal.slack.com/archives/C031USXS2FJ/p1729521714992659) to implement it like this. Per rbean, the secrets are in the prod rh01, prod p01, and prod p02 clusters.

Reply:
Yeah, ok, the secrets are there, but no RPAs have the required settings yet, right? So that should be done first, otherwise this will break users.

Reply:
Sure, I am not so well versed in this, that's why I opened the discussion before. What do you think needs to be done now?

Reply:
Now I noticed that there is `updateComponentSBOM` and it defaults to false. So ok, without any change to RPA, this will not happen. But in that case `.atlas.server` should definitely not be required to be in the data file. If you don't want to have a default for the server, then the reading of the server (and throwing an error if missing) should only happen if updateComponentSBOM is true.

Also, I noticed that updateComponentSBOM is emitted as a result, but that result is not used for anything. So what is it for? The other two tasks should probably have a `when:` clause and be executed only if the updateComponentSBOM result is true, no?

Reply:
The smoothest solution probably is:
* The Atlas field in the RPA is fully optional.
* Only enable product-level SBOM creation and push when the Atlas field is present.
* Only enable component-level SBOM update when the Atlas field is present and the enabling flag is set to true.

What do you think?

The updateComponentSBOM result flag will be used with the #656 changes once that is merged.

Reply:
I wasn't aware there were two different things - product-level SBOM and component-level SBOMs.

I think both of them should be treated equally, no? But It seems this PR only adds tasks to create and upload product sbom, not component-level sboms.

> In the latest commit, I made the SBOM processing optional when the Atlas fields are not provided in the RPA.

Yeah, that looks good.

Reply:
I implemented it in the last commit, see if you feel it's alright.

Reply:
Yeah, it looks ok. But we don't have that other update task yet, so I will have to review again when that gets added.

Reply:
Sorry if it was confusing, merging #656 is taking kind of long so I though I would expedite the process here by first putting product SBOMs for review and the component stuff. In hindsight I probably should have created two PRs.

Reply:
No worries. One reason it takes longer is that we had our face to face meeting last week, so I responsiveness was limited.

Reply:
You already said it, but asking again just in case... If you just set the Atlas secret, product sbom will be uploaded, but not component sboms. Component sboms will be created from original sboms and then uploaded only if `updateComponentSBOM` is set. What is the scenario for this? Why would you want the product sbom, but not the component sboms in Atlas?

Also, isn't it confusing from the user point of view to call the flag `updateComponentSBOM`? Because when missing/set to false, you will have no component sboms in Atlas at all, right? So I'd say it would make more sense to call it `uploadComponentSBOM`.

Reply:
After some discussion, we can probably leave that flag out. 

The renaming also makes sense.

Reply:
Awesome, less things is always better (well, most of the time) :)